src/SFA.DAS.ApplyService.Data/ApplyRepository.cs

```diff
@@ -796,5 +980,95 @@ public async Task<string> GetNextRoatpApplicationReference()
                 return $"APR{nextInSequence}";
             }
         }
+
+        public async Task<bool> RecordFinancialGrade(Guid applicationId, FinancialReviewDetails financialReviewDetails, string financialReviewStatus)
+        {
+            using (var connection = new SqlConnection(_config.SqlConnectionString))
+            {
+                try
+                {
+                    var result = await connection.QueryAsync<ApplyData>(@"SELECT ApplyData FROM Apply WHERE ApplicationId = @applicationId",
```

Comment:
More of a comment but I'd have personally used GetApplication() than have a separate Dapper query. That's just so retrieving application info is done in one place, but then you can argue that your way brings back a lighter object

Reply:
Yeah, I did try and do the update in a single step here using JSON_MODIFY but found that it set the field value to serialised JSON rather than an object, it seems that only works for simple types.